server/src/server.ts

```diff
@@ -588,16 +520,14 @@ export class Server {
 			const wordAndRange = autoCompleteResponses.wordAndRange;
 			// Don't send requests for very short words
 			if (wordAndRange.word.length < 2) return [];
-			const autoCompletionRequest: AutocompleteRequest = {
-				"query": "autocomplete",
-				"args": {
-					"partial-symbol": wordAndRange.word,
-					"context": "code"
-				}
-			};
-			this.sendRequestForDocument(doc, autoCompletionRequest, this.configurationSettings.flyCheck ? "lax" : undefined);
+			const lax = this.configurationSettings.flyCheck ? "lax" : undefined;
+			const fstar_conn = this.getFStarConnection(doc, lax);
+			// TODO(klinvill): should we await the response here? The autocomplete response table is populated asynchronously.
```

Comment:
Eventually we want to drop the response table entirely.  See #28.  Then we can just do `autoCompleteResponses = await fstar_conn.autocompleteRequest(...)` at the beginning of the function.

Reply:
I'll just leave this for a future PR since it's already documented as an issue.